pubsub.go

```diff
@@ -170,6 +170,11 @@ type PubSub struct {
 	protoMatchFunc ProtocolMatchFn
 
 	ctx context.Context
+
+	// appSpecificRpcInspector is an auxiliary that may be set by the application to inspect incoming RPCs prior to
+	// processing them. The inspector is invoked on an accepted RPC right prior to handling it.
+	// The return value of the inspector function is a boolean indicating whether the RPC should be processed or not.
+	appSpecificRpcInspector func(peer.ID, *RPC) bool
```

Comment:
lets return a loggable error to provide some context.

Reply:
https://github.com/libp2p/go-libp2p-pubsub/pull/509/commits/776d8592b871718e6f63228f5ce8f9c919f941ea